packages/ember-debug/tests/main_test.js

```diff
@@ -105,14 +105,12 @@ QUnit.test('Ember.deprecate throws deprecation if second argument is falsy', fun
   });
 });
 
-QUnit.test('Ember.deprecate does not throw deprecation if second argument is a function and it returns true', function() {
+QUnit.test('Ember.deprecate throws deprecation if second argument is a function and it returns true', function() {
   expect(1);
 
-  Ember.deprecate('Deprecation is thrown', function() {
-    return true;
-  }, { id: 'test', until: 'forever' });
-
-  ok(true, 'deprecation was not thrown');
+  throws(() => {
```

Comment:
It doesn't work when using `expectDeprecation` seems like it doesn't even call invoke.